mypy/test/test_check.py

```diff
@@ -122,13 +122,12 @@ def run_test_once(self, testcase: DataDrivenTestCase, incremental=0) -> None:
             a = e.messages
         a = normalize_error_messages(a)
 
-        if output != a and mypy.myunit.UPDATE_TESTCASES:
+        if output != a and pytest.config.getoption('UPDATE_TESTCASES'):
             update_testcase_output(testcase, a, mypy.myunit.APPEND_TESTCASES)
 
-        assert_string_arrays_equal(
-            output, a,
+        assert output == a, \
             'Invalid type checker output ({}, line {})'.format(
-                testcase.file, testcase.line))
+                testcase.file, testcase.line)
```

Comment:
How does the output from this turn out when there's a mismatch?

The `assert_string_arrays_equal` helper is pretty nice for comparing things that look like the error output from a typechecker -- I'd want to make sure we get something equally nice after the transition. The most essential feature might be just the fact that it prints both actual and expected as sequences of lines.